docs/Users/Tutorials/External_Bears.rst

```diff
@@ -0,0 +1,521 @@
+External Bears
+==============
+
+Welcome. This tutorial will teach you how to use the
+``@external_bear_wrap`` decorator in order to write Bears in languages other
+than Python.
+
+.. note::
+
+  This tutorial assumes that you already know the basics. If you are new, please
+  refer to the
+  :doc:`Writing Bears<Writing_Bears>` section.
+
+  If you are planning to create a bear that uses an already existing tool (aka
+  linter), please refer to the
+  :doc:`Linter Bears<Linter_Bears>` section.
+
+Why is This Useful?
+-------------------
+
+coala is a great tool if you want to write static code analysis without having
+to worry about the interaction with the user. That makes it an excellent choice
+for researchers and other people who want to build new static analysis routines.
+Enabling developers to write external Bears means that they do not need to waste
+any time learning new languages and they can get faster to the interesting part,
+**static code analysis**.
+
+How Does This Work?
+-------------------
+
+By using the ``@external_bear_wrap`` decorator you will have all the necessary
+data sent to your external executable (filename, lines, settings) as a JSON
+string via ``stdin``. Afterwards the analysis takes place in your executable
+that can be written in **literally** any language. In the end you will have to
+provide the ``Results`` in a JSON string via stdout.
+
+In this tutorial we will go through 2 examples where we create a very simple
+bear. The first example will use a **compiled** language that creates a
+standalone binary, ``C++``, whilst in the second example we will take a look at
+``JS`` that needs ``node`` in order to run out of the browser.
+
+External Bear Generation Tool
+-----------------------------
+
+If you really do not want to write any Python code, there is a tool
+`here <https://gitlab.com/coala/coala-bear-management>`__,
+``coala-bears-create``, that will create the wrapper for you. We will be using
+
+::
+
+    $ coala-bears-create -ext
+
+in order to generate the wrapper for the bear.
+
+Writing a Bear in C++
+---------------------
+
+The bear that will be created with this tutorial will check whether there is any
+**coala** spelled with a capital ``C`` since that is a horrible mistake for one
+to make.
+
+1. Create a new directory and make it your current working directory.
+2. Run ``coala-bears-create`` as mentioned above in order to create the wrapper
+   for our ``C++`` bear and hit ``Enter`` for the first question. You can skip
+   this step and use the wrapper provided below.
+3. Name the bear ``coalaCheckBear`` and the executable ``coalaCheck_cpp``.
+4. Reply to the rest of the questions (hit ``Enter`` at settings). After the
+   script is finished running there should be 2 files in your current directory:
+   ``coalaCheckBear.py`` (the wrapper) and ``coalaCheckBearTest.py``.
```

Comment:
What do they need to reply. Do they simply have to go with the defaults? Because I don't think first time bear writers would understand this well.